context.go

```diff
@@ -11,6 +11,7 @@ import (
 	"runtime"
 	"strings"
 
+	"fmt"
```

Comment:
nit: could move this import up with the other std lib imports

LGTM besides, unless there should be some new test cases as well.